op-devstack/dsl/l2_op_rbuilder.go

```diff
@@ -55,6 +58,34 @@ func (el *OPRBuilderNode) Start() {
 	el.control.OPRBuilderNodeState(el.inner.ID(), stack.Start)
 }
 
-func (el *OPRBuilderNode) FlashblocksClient() *FlashblocksWSClient {
-	return NewFlashblocksWSClient(el.inner.FlashblocksClient())
+// listenForWS reads from the given websocket client for the specified duration,
+// and forwards any received messages to the output channel.
+func listenForWS(ctx context.Context, logger log.Logger, wsClient *opclient.WSClient, duration time.Duration, output chan<- []byte, done chan<- struct{}) error {
```

Comment:
This looks like the same method we're implementing in the flashblocks AT utils.go. In wonder if `WSClient` should have a `ReadAll` method that reads until the context it is passed is done and it sends received bytes to a channel like this. Then the client could also ensure the listening thread has fully stopped as part of its `Close` method.

Reply:
Added, allowing me to delete this util.